src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1020,49 +1019,101 @@ public void stopDetector(
         }));
     }
 
-    private void resetLatestHistoricalDetectorTaskState(
+    private <T> void resetLatestHistoricalDetectorTaskState(
```

Comment:
this method may also reset realtime tasks state.  Could you change the name?

Reply:
good catch, will change